crates/cheatcodes/src/test/expect.rs

```diff
@@ -424,38 +471,42 @@ pub(crate) fn handle_expect_emit(state: &mut Cheatcodes, log: &alloy_primitives:
         .expect("we should have an emit to fill or check");
 
     let Some(expected) = &event_to_fill_or_check.log else {
-        // Fill the event.
-        event_to_fill_or_check.log = Some(log.data.clone());
+        // Unless the caller is trying to match an anonymous event, the first topic must be
+        // filled.
+        // TODO: failing this check should probably cause a warning
```

Comment:
I think we'd want to revert here ideally, should be possible after https://github.com/bluealloy/revm/pull/1610